File: app/src/main/java/com/pkfcooperparry/template/ui/splash/SplashActivity.java
package com.pkfcooperparry.template.ui.splash;

import android.os.Bundle;
import android.os.Handler;

import com.pkfcooperparry.template.BR;
import com.pkfcooperparry.template.R;
import com.pkfcooperparry.template.databinding.ActivitySplashBinding;
import com.pkfcooperparry.template.ui.base.BaseActivity;
import com.pkfcooperparry.template.utils.Navigator;

import javax.inject.Inject;


public class SplashActivity extends BaseActivity<ActivitySplashBinding, SplashViewModel> implements SplashViewInterface {
    private static final int SPLASH_DISPLAY_LENGTH = 1000;
    @Inject
    SplashViewModel splashViewModel;

    @Inject
    Navigator navigator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        splashViewModel.setViewInterface(this, this);
        Handler handler = new Handler();
        handler.postDelayed(() -> splashViewModel.decideNextActivity(), SPLASH_DISPLAY_LENGTH);
    }


    @Override
    public int getBindingVariable() {
        return BR.viewModel;
    }


    @Override
    public SplashViewModel getViewModel() {
        return splashViewModel;
    }

    @Override
    protected String getActionBarTitle() {
        return null;
    }

    @Override
    protected int getActionBarType() {
        return KEY_NO_ACTION_BAR;
    }

    @Override
    public int getLayoutId() {
        return R.layout.activity_splash;
    }


    @Override
    public void openLoginActivity() {
        navigator.navigateToSignInView(this);
    }

    @Override
    public void openMainActivity() {
        navigator.navigateToMainView(this);
    }
}

File: app/src/main/java/com/pkfcooperparry/template/ui/splash/SplashActivity.kt
package com.pkfcooperparry.template.ui.splash

import android.os.Bundle
import android.os.Handler

import com.pkfcooperparry.template.BR
import com.pkfcooperparry.template.R
import com.pkfcooperparry.template.databinding.ActivitySplashBinding
import com.pkfcooperparry.template.ui.base.BaseActivity
import com.pkfcooperparry.template.utils.Navigator

import javax.inject.Inject


class SplashActivity : BaseActivity<ActivitySplashBinding, SplashViewModel>(), SplashViewInterface {

    override var viewModel: SplashViewModel? = null
        @Inject set

    var navigator: Navigator? = null
        @Inject set

    override val bindingVariable: Int
        get() = BR.viewModel

    protected override val actionBarTitle: String?
        get() = null

    protected override val actionBarType: Int
        get() = BaseActivity.KEY_NO_ACTION_BAR

    override val layoutId: Int
        get() = R.layout.activity_splash

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel!!.setViewInterface(this, this)
        val handler = Handler()
        handler.postDelayed({ viewModel!!.decideNextActivity() }, SPLASH_DISPLAY_LENGTH.toLong())
    }


    override fun openLoginActivity() {
        navigator!!.navigateToSignInView(this)
    }

    override fun openMainActivity() {
        navigator!!.navigateToMainView(this)
    }

    companion object {
        private val SPLASH_DISPLAY_LENGTH = 1000
    }
}
